In [11]:
from utils.llm.mistral import MistralLLM
from PyPDF2 import PdfReader

In [3]:
mistral = MistralLLM()

In [7]:
import os
os.getcwd()

'/model'

In [37]:
pdf_path = "3_link_prediction/KGQA.pdf"
pdf = PdfReader(pdf_path)
diss_text = ""
for page in pdf.pages[39:43]:
    diss_text += page.extract_text()
    diss_text = diss_text.replace("\n", " ")

with open("3_link_prediction/KGQA.txt", "w") as f:
    f.write(diss_text)

In [34]:
%load_ext autoreload
%autoreload 2

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [38]:
response = mistral("""
Summarise the results on different types of questions in section 5.3. \n\n""" + diss_text)
display(response)

'In section 5.3, the results of different types of questions are discussed, focusing on the ComplexWebQuestions (CWQ) dataset. The CWQ dataset consists of four types of questions: Comparative, Composition, Conjunction, and Superlative.\n\n1. Comparative: What country with an ISO less than 233 borders Russia?\n2. Composition: What actress plays Claire on a television show with the theme song ’Lonely Girl’?\n3. Conjunction: In what country is Arabic spoken and is the birthplace of Barbara Starr?\n4. Superlative: From all the sights in Madrid, what is the latest exhibition venue that opened in that city?\n\nThe results show that the performance of the models varies across different question types. Here are the average performances for each type of question and the overall average:\n\n| Method | Comparative | Composition | Conjunction | Superlative | Average |\n|--------|-------------|-------------|-------------|-------------|---------|\n| Baseline | 0.150       | 0.177        | 0.213     

In section 5.3, the results of different types of questions are discussed, focusing on the ComplexWebQuestions (CWQ) dataset. The CWQ dataset consists of four types of questions: Comparative, Composition, Conjunction, and Superlative.\n\n1. Comparative: What country with an ISO less than 233 borders Russia?\n2. Composition: What actress plays Claire on a television show with the theme song ’Lonely Girl’?\n3. Conjunction: In what country is Arabic spoken and is the birthplace of Barbara Starr?\n4. Superlative: From all the sights in Madrid, what is the latest exhibition venue that opened in that city?\n\nThe results show that the performance of the models varies across different question types. Here are the average performances for each type of question and the overall average:\n\n| Method | Comparative | Composition | Conjunction | Superlative | Average |\n|--------|-------------|-------------|-------------|-------------|---------|\n| Baseline | 0.150       | 0.177        | 0.213        | 0.212        | 0.188    |\n| Perfect-Path | 0.283       | 0.576        | 0.430        | 0.423        | 0.428    |\n| BM25-n | 0.164       | 0.151        | 0.473        | 0.113        | 0.225    |\n| S-BERT-n | 0.262       | 0.269        | 0.275        | 0.170        | 0.244    |\n| ToG-LP-4 | 0.248       | 0.111        | 0.157        | 0.132        | 0.162    |\n\n

The Perfect-Path method performs the best on Comparative questions, while it excels on Composition questions. BM25-n performs relatively well on Composition questions, but its performance is poor on Comparative, Conjunction, and Superlative questions. S-BERT-n performs well on Comparative and Composition questions, but its performance is poor on Conjunction and Superlative questions. ToG-LP-4 performs relatively well on Comparative questions but underperforms on Composition, Conjunction, and Superlative questions.\n\nIt is worth noting that the class balance when selecting a subset of questions is not discussed in the provided text. However, it is essential to consider class balance when evaluating the performance of models on imbalanced datasets, as it can significantly impact the results.